homeassistant/components/qnap_qsw/binary_sensor.py

```diff
@@ -69,12 +129,22 @@ def __init__(
         coordinator: QswDataCoordinator,
         description: QswBinarySensorEntityDescription,
         entry: ConfigEntry,
+        type_id: int | None = None,
     ) -> None:
         """Initialize."""
-        super().__init__(coordinator, entry)
-        self._attr_name = f"{self.product} {description.name}"
+        super().__init__(coordinator, entry, type_id)
+        if type_id is not None:
+            if description.qsw_type == QswEntityType.LACP_PORT:
+                entity_str = f" LACP Port {type_id} "
```

Comment:
This looks like high risk of a future refactoring error. Please refactor to not tie the name and the unique id or add a test for the unique ids.

Reply:
Ok, I will try to refactor as you suggest, but I'm not sure I've got your point though...
Do you want me to set name and unique_id separately inside of the "if/else" port type check? (removing the aux str variable)

Reply:
@bdraco done in 89b6c2d51d905dbf4fc26041c50ab2a3f6d2f2a2